src/gmt_init.c

```diff
@@ -9443,22 +9569,44 @@ GMT_LOCAL int gmtinit_loaddefaults (struct GMT_CTRL *GMT, char *file) {
 		keyword[0] = value[0] = '\0';	/* Initialize */
 		sscanf (line, "%s = %[^\n]", keyword, value);
 
-		if (gmtlib_setparameter (GMT, keyword, value, false))
+		if (gmtlib_setparameter (GMT, keyword, value, true))
 			error++;
 		else {
 			int case_val = gmt_hash_lookup (GMT, keyword, keys_hashnode, GMT_N_KEYS, GMT_N_KEYS);
 			if (case_val >= 0) GMT_keywords_updated[case_val] = true;		/* Leave a record that this keyword is no longer a default one */
 		}
 	}
-
 	fclose (fp);
+
+	error += gmtinit_update_theme (GMT);	/* If we got a GMT_THEME setting, take delayed action now */
+
 	gmtinit_verify_encodings (GMT);
 
 	if (error) gmt_message (GMT, "%d GMT Defaults conversion errors in file %s!\n", error, file);
 
 	return (GMT_NOERROR);
 }
 
+GMT_LOCAL int gmtinit_update_theme (struct GMT_CTRL *GMT) {
+	int error = GMT_NOERROR;
+	char theme_file[PATH_MAX] = {""};
+
+	if (!GMT->current.setting.update_theme) return GMT_NOERROR;	/* Nothing to do */
+
+	/* Got a GMT_THEME setting, take delayed action now */
+	GMT->current.setting.update_theme = false;
+	if (!strcmp (GMT->current.setting.theme, "classic"))	/* Just reload the classic defaults */
+		gmtinit_conf_classic (GMT);
+	else if (!strcmp (GMT->current.setting.theme, "modern"))
+		gmtinit_conf_modern (GMT);
```

Comment:
It seems like the modern theme is hard-coded in through gmtinit_conf_modern and gmtinif_conf_modern_override. Is this interpretation correct? Is there any functionality to the /share/themes/modern.conf file?

Reply:
The thought was that we wanted modern to be the default in modern mode.  Relying on people to read up on docs and change via GMT_THEME will not lead to much adoption.  Given modern mode is young enough for us to make a somewhat radical change I say we do it that way.  The oldtimers probably are not using modern mode anyway and will be unaffected.

Reply:
I think that's a good choice. My concern is that modern.conf and the defaults provided in gmtinit_conf_modern_override are redundant. I expect that other people could run into a similar situation as I did in changing modern.conf and being surprised when nothing happened. Perhaps we could remove the modern.conf file and allow users to get the modern defaults through ```gmt defaults -D ``` in modern mode?

Reply:
True.  I think this was partly because it was a branch and for testing, so I kept a separate modern.conf file in addition to the gmtinit_conf_modern_override function.  I am not sure if the @GenericMappingTools/core will see this all the same way: I am OK with getting rid of the modern.conf file and let those in the classic remain classic, but it also seems a bit hash to force die-hard classicist to recreate modern.conf from what they see in gmt defaults -D.  So maybe more input from the developers on how they see this is fine at this point

Reply:
I am expecting an easy way of using the classic theme under modern. It has to exist, otherwise we could not use other themes on modern.
From my side there two things I like on modern theme and two that maybe not
- above all auto scale annotations and a thinner frame line for psxy
- Highly doubtful on letting go the pajama frames on Mercator maps, and wait to see about new default fonts.

Reply:
I can understand the dilemna around modern.conf. At least, I think it should be consistent with a classic.conf (which was mentioned early on) or neither.

My impression from reading the earlier discussion was that people were generally on board with modern MAP_FRAME_TYPE = plain. Was that an incorrect interpretation, @joa-quim?

Reply:
Well, we talked a lot about fonts but very little on MAP_FRAME_TYPE (at least I didn't give it much attention). But now, more I think on it less I like the idea of *all maps are plain*.

Honestly, if we want to come up with a modern look we should implement themes like in https://www.datanovia.com/en/blog/ggplot-themes-gallery/